### PR TITLE
update on scaler limited to Max

### LINKED ISSUE #2297: FleetAutoscaler has confusing ScalingLimited warning when scaling down
**What happened**:
I got a warning:
  Warning  ScalingLimited    2m53s (x14 over 4h47m)  fleetautoscaler-controller  Scaling fleet my-fleet was limited to maximum size of 3

**What you expected to happen**:
My fleet is configured with a minimum of 3 replicas and a max of 99. I think the error is supposed to say:

Scaling fleet my-fleet was limited to minimum size of 3

**How to reproduce it (as minimally and precisely as possible)**:
Run a FleetAutoscaler and let it try to scale up and down

**Anything else we need to know?**:

**Environment**:
- Agones version: 1.17.0
- Kubernetes version (use `kubectl version`): 1.20
- Cloud provider or hardware configuration: GKE
- Install method (yaml/helm): helm
- Troubleshooting guide log(s):
- Others:

### PR DESCRIPTION
**Pull request by SaitejaTamma**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes  #2297

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Looks like you failed linting!

If you want to locally test linting, you can run `make lint`.

**Review by markmandel — COMMENTED** [review]
Would be good to unit test this to ensure it (a) does what we expect and (b) we don't break it in the future.

**Comment by markmandel** [comment]
```
Error: 14 UNAVAILABLE: Connection dropped
    at Object.callErrorFromStatus (/go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/call.js:31:26)
    at Object.onReceiveStatus (/go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/client.js:327:49)
    at Object.onReceiveStatus (/go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:299:181)
    at /go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/call-stream.js:145:78
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
Emitted 'error' event on ClientReadableStreamImpl instance at:
    at Object.onReceiveStatus (/go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/client.js:327:28)
    at Object.onReceiveStatus (/go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:299:181)
    at /go/src/agones.dev/agones/sdks/nodejs/node_modules/@grpc/grpc-js/build/src/call-stream.js:145:78
    at processTicksAndRejections (internal/process/task_queues.js:79:11) {
  code: 14,
  details: 'Connection dropped',
  metadata: Metadata { internalRepr: Map {}, options: {} }
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ testSDK: `node ./testSDKClient.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ testSDK script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-01-25T20_02_53_125Z-debug.log
make[2]: *** [includes/sdk.mk:145: run-sdk-conformance-no-build] Error 1
make[1]: *** [includes/sdk.mk:153: run-sdk-conformance-test] Error 2
make: *** [includes/sdk.mk:159: run-sdk-conformance-test-node] Error 2
```

Node failing conformance is pretty rare.

File: pkg/fleetautoscalers/controller.go [hunk 1]
```diff
@@ -366,7 +366,12 @@ func (c *Controller) updateStatus(ctx context.Context, fas *autoscalingv1.FleetA
 
 	if !apiequality.Semantic.DeepEqual(fas.Status, fasCopy.Status) {
 		if scalingLimited {
-			c.recorder.Eventf(fas, corev1.EventTypeWarning, "ScalingLimited", "Scaling fleet %s was limited to maximum size of %d", fas.Spec.FleetName, desiredReplicas)
+			// scalingLimited indicates that the calculated scale would be above or below the range defined by MinReplicas and MaxReplicas
+			if currentReplicas > desiredReplicas {
+				c.recorder.Eventf(fas, corev1.EventTypeWarning, "ScalingLimited", "Scaling fleet %s was limited to maximum size of %d", fas.Spec.FleetName, desiredReplicas)
+			} else {
+				c.recorder.Eventf(fas, corev1.EventTypeWarning, "ScalingLimited", "Scaling fleet %s was limited to minimum size of %d", fas.Spec.FleetName, desiredReplicas)
+			}
```
Comment (markmandel) [inline pkg/fleetautoscalers/controller.go:374]:
```suggestion
      msg := "Scaling fleet %s was limited to minimum size of %d"
			if currentReplicas > desiredReplicas {
         msg := "Scaling fleet %s was limited to maximum size of %d"
			} 
			
			c.recorder.Eventf(fas, corev1.EventTypeWarning, "ScalingLimited", msg, fas.Spec.FleetName, desiredReplicas)
```

WDYT? Remove some duplication of strings?

Comment (SaitejaTamma) [reply pkg/fleetautoscalers/controller.go:374]: Yes make sense

**Review by markmandel — APPROVED** [review]

**Pull request merged**